### PR TITLE
homework-6

### PR DESCRIPTION
**Pull request by tribaeka**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,3 +5,4 @@ DB_PASS=postgres
 DB_PORT=5438
 DB_HOST=localhost
 ENABLE_SEQUELIZE_LOGGING: false
+JWT_SECRET=secret
```
Comment (kuzanatoliy) [inline .env:8]: Would be nice to use environment

File: homework-2/index.ts [hunk 2]
```diff
@@ -23,10 +24,12 @@ const app: Application = express();
 const serverInstance = createServer(app);
 const { APP_PORT } = process.env;
 
+app.use(cors())
```
Comment (kuzanatoliy) [inline homework-2/index.ts]: would be nice to use more specific cors rules

File: homework-2/middewares/authMiddlewares.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    password: string;
+}
+
+const { JWT_SECRET } = process.env;
+
+export async function login(req: Request<unknown, unknown, IAuthReqBody>, res: Response, next: NextFunction): Promise<void> {
+    const { login, password } = req.body;
+    const user = await UserService.getUserByLogin(login);
+
+    if (user === undefined || user.password !== password || user.isDeleted) {
```
Comment (kuzanatoliy) [inline homework-2/middewares/authMiddlewares.ts:19]: What do you think to create specific request to DB that would check this variables?

File: homework-2/middewares/authMiddlewares.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        next(new HttpError(httpStatus["403_NAME"], httpStatus["403_MESSAGE"], httpStatus.FORBIDDEN));
+    } else {
+        const payload = { sub: user.id, login: user.login };
+        const token = sign(payload, JWT_SECRET as string, { expiresIn: '1m' });
+        res.send(token);
+    }
+}
+
+export function protectedMiddleware(req: Request, res: Response, next: NextFunction): void {
+    const token = req.headers['x-access-token'] as string;
```
Comment (kuzanatoliy) [inline homework-2/middewares/authMiddlewares.ts]: Please move header name to constants

File: package.json [hunk 5]
```diff
@@ -21,12 +21,16 @@
   "author": "Yahor Hlushak <Yahor_Hlushak@epam.com>",
   "license": "ISC",
   "dependencies": {
+    "@types/cors": "2.8.10",
+    "@types/jsonwebtoken": "^8.5.1",
```
Comment (kuzanatoliy) [inline package.json]: Please use fixed versions

File: .env (hunk 1 above)
Comment (tribaeka) [reply .env:8]: This variable located in the .env file and accessed via process.end so I think I'm using environment here. Can you clarify, please?

File: homework-2/index.ts (hunk 2 above)
Comment (tribaeka) [reply homework-2/index.ts]: Done

File: homework-2/middewares/authMiddlewares.ts (hunk 3 above)
Comment (tribaeka) [reply homework-2/middewares/authMiddlewares.ts:19]: Good idea, I have implemented in `getAuthenticatedUser` function.

File: homework-2/middewares/authMiddlewares.ts (hunk 4 above)
Comment (tribaeka) [reply homework-2/middewares/authMiddlewares.ts]: Fixed

File: package.json (hunk 5 above)
Comment (tribaeka) [reply package.json]: Fixed thanks

**Pull request merged**